File: 엄마호랑이반/7주차 문제/최혜령/신고결과받기.py
def solution(id_list, report, k):
    ids = {}
    reports = {}
    for elem in id_list:
        ids[elem] = 0
        reports[elem] = []
    
    
    for elem in report:
        reporter, reported = elem.split(" ")
        if reporter not in reports[reported]:
            reports[reported].append(reporter)
    
    for key, value in reports.items():
        if len(value) >= k:
            for elem in value:
                ids[elem] += 1
    
    answer = list(ids.values())
        
    return answer

File: 엄마호랑이반/7주차 문제/이다영/신고결과받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        
        Map<String, Integer> userId = new HashMap<>();
        List<Set<String>> reporter = new ArrayList<>();
        for (int i = 0; i < id_list.length; i++) {
            userId.put(id_list[i], i);
            reporter.add(new HashSet<>());
        }
        
        for (String r : report) {
            String[] tmp = r.split(" ");
            String er = tmp[0]; // 신고자
            String ed = tmp[1]; // 신고당한자
            
            int idx = userId.get(ed);
            reporter.get(idx).add(er);
        }
        
        for (int i = 0; i < id_list.length; i++) {
            if(reporter.get(i).size() >= k) {
                for (String person : reporter.get(i)) {
                    int idx = userId.get(person);
                    answer[idx] += 1;
                }
            }
        }
                
        
        return answer;
    }
}
